test/test_grid.py

```diff
@@ -778,3 +781,57 @@ def test_build_face_edges_connectivity_fillvalues(self):
         self.assertTrue(
             np.array_equal(res_face_nodes_connectivity,
                            uds._ds["Mesh2_face_nodes"].values))
+
+
+class TestBallTree(TestCase):
+
+    corner_grid_files = [gridfile_CSne30, gridfile_mpas]
+    center_grid_files = [gridfile_mpas]
+
+    def test_construction_from_corner_nodes(self):
+        """Tests the construction of the ball tree on corner nodes and performs
+        a sample query."""
+
+        for grid_file in self.corner_grid_files:
+            uxgrid = ux.open_grid(grid_file)
+
+            # performs a sample query
+            d, ind = uxgrid.corner_node_balltree.query([3.0, 3.0], k=3)
```

Comment:
The queries are all done in lat/lon, do we not really need to worry about cartesian coordinates? I can't think of a use case that would need cartesian coordinates but I figure I would ask.

Reply:
both the `corner_node_balltree` and `center_node_balltree` are constructed using lat/lon coordinates, and expect lat/lon inputs. We could consider adding a tree that is build from and queries using Cartesiain, but right now it's only lat/lon